ci/travis/ci.sh

```diff
@@ -53,16 +61,33 @@ preload() {
   local variable_definitions
   variable_definitions=($(python "${ROOT_DIR}"/determine_tests_to_run.py))
   if [ 0 -lt "${#variable_definitions[@]}" ]; then
-    export "${variable_definitions[@]}"
+    local expression
+    expression="$(printf "%q " "${variable_definitions[@]}")"
+    eval "${expression}"
+    printf "%s\n" "${expression}" >> ~/.bashrc
   fi
 
   if ! (set +x && should_run_job ${job_names//,/ }); then
+    if [ -n "${GITHUB_WORKFLOW-}" ]; then
+      # If this job is to be skipped, emit an 'exit' command into .bashrc to quickly exit all following steps.
```

Comment:
i'm pretty sure github action has some feature like this? fail fast or something?

Reply:
I don't think it does? And I think you mean `continue-on-error`? Fail-fast means that if a job fails then the entire workflow fails, but here we want it to succeed. `continue-on-error` would make it succeed but errors can occur due to lots of other issues, which we wouldn't want to ignore.

Reply:
If `continue-on-error` is set to false, all the following steps will be skipped automatically right?

Reply:
I thought it only skips continues this step if there's an error and then it moves onto the next one. But regardless, I'm saying the bigger issue is something else: using it would mask _all_ errors that step, which we really don't want. We only want to exit the build job under a very particular set of conditions, which is that some flags determine that the job should be skipped. This seemed like the easiest way of achieving that without having to put a lot of boilerplate in every CI's configuration files.